### PR TITLE
Add test for email_settings model

### PR DESCRIPTION
**Pull request by sonalkr132**

### CODE REVIEW

File: spec/models/email_settings_spec.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    context 'conference has cfp' do
+      before do
+        conference.program.update_attributes(cfp: create(:cfp,
+                                                         start_date: Date.new(2014, 04, 29),
+                                                         end_date: Date.new(2014, 05, 06)))
+      end
+
+      it 'returns hash with cfp start and end date' do
+        cfp_dates_hash = { 'cfp_start_date' => Date.new(2014, 04, 29), 'cfp_end_date' => Date.new(2014, 05, 06) }
```
Comment (ChrisBr) [inline spec/models/email_settings_spec.rb]: You can move this also to the before block 
``
cfp_dates_hash = { 'cfp_start_date' => Date.new(2014, 04, 29), 'cfp_end_date' => Date.new(2014, 05, 06) }
expected_hash.merge!(cfp_dates_hash)
``

File: spec/models/email_settings_spec.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        expect(conference.email_settings.get_values(conference, user)).to eq expected_hash
+      end
+    end
+
+    context 'conference has venue' do
+      before { conference.update_attributes(venue: create(:venue)) }
+
+      it 'returns hash with venue and venue_address' do
+        venue_hash = { 'venue' => 'Suse Office', 'venue_address' => 'Maxfeldstrasse 5, Nuremberg, Germany' }
+        expected_hash.merge!(venue_hash)
```
Comment (ChrisBr) [inline spec/models/email_settings_spec.rb]: Same here

File: spec/models/email_settings_spec.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    context 'conference has registration period' do
+      before do
+        conference.update_attributes(registration_period: create(:registration_period,
+                                                                 start_date: Date.new(2014, 05, 03),
+                                                                 end_date: Date.new(2014, 05, 05)))
+      end
+
+      it 'returns hash with registration_start_date and registration_end_date' do
+        registration_period_hash = { 'registration_start_date' => Date.new(2014, 05, 03), 'registration_end_date' => Date.new(2014, 05, 05) }
```
Comment (ChrisBr) [inline spec/models/email_settings_spec.rb]: Same here

File: spec/models/email_settings_spec.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
+require 'spec_helper'
+
+describe EmailSettings do
+  let(:conference) { create(:conference, short_title: 'goto', start_date: Date.new(2014, 05, 01), end_date: Date.new(2014, 05, 06)) }
```
Comment (ChrisBr) [inline spec/models/email_settings_spec.rb]: Why do you need to set :shorttitleand start/end_date?

File: spec/models/email_settings_spec.rb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  end
+
+  describe '#generate_event_mail' do
+    let(:event_template) do
+      "Dear {name}\n\nWe are very pleased" \
+      'to inform you that your submission {eventtitle} has been accepted for the conference {conference}.'
+    end
+
+    it 'replaces fillers in template' do
+      expected_text = "Dear John Doe\n\nWe are very pleased" \
```
Comment (ChrisBr) [inline spec/models/email_settings_spec.rb]: Please test also if no name is set as we just had this regression currently!

File: spec/models/email_settings_spec.rb (hunk 4 above)
Comment (sonalkr132) [inline spec/models/email_settings_spec.rb]: `:shortitle` is set by sequence in factory. I don't want it to change between tests cause I am using it in expected hash:
```Ruby
'registrationlink' => 'http://localhost:3000/conference/goto/register',
 'conference_splash_link' => 'http://localhost:3000/conference/goto',
 'schedule_link' => 'http://localhost:3000/conference/goto/schedule',
```
`:stard/end_date` is set by:
```Ruby
start_date { Date.today }
end_date { 6.days.from_now }
```
This one although being same in every test run, depends on time zone. Check out this [failing travis build](https://travis-ci.org/sonalkr132/osem/builds/118696738#L3055). It was 7 May on travis server when it was 6 May on my PC.

File: spec/models/email_settings_spec.rb (hunk 5 above)
Comment (sonalkr132) [inline spec/models/email_settings_spec.rb]: @ChrisBr  Thanks for bringing it up. I have added this test [here](https://github.com/openSUSE/osem/pull/917/files#diff-4dfcd74d95dd9208044a4c1f32b3f24bR36). `get_values` is responsible for setting the hash, `generate_event_mail` merely uses the value set by `get_values`. Do you think I should test it in `generate_event_mail` too?

**Comment by ChrisBr** [comment]
LGTM

**Pull request merged**